cheroot/ssl/builtin.py

```diff
@@ -200,4 +199,5 @@ def env_dn_dict(self, env_prefix, cert_value):
 
     def makefile(self, sock, mode='r', bufsize=DEFAULT_BUFFER_SIZE):
         """Return socket file object."""
-        return MakeFile(sock, mode, bufsize)
+        cls = StreamReader if 'r' in mode else StreamWriter
```

Comment:
It feels like this logic duplicates logic in makefile.MakeFile (Python 3 version). Ideally, this decision would be made in one place. Though, if one of these methods is deprecated, that's fine.

Reply:
I decided to delay this refactoring. I also don't like the inability to remove `MakeFile` from `cheroot.server` module completely...

Reply:
Oh, and I'm also trying hard to decouple those things.

Reply:
Which might mean removing makefile in lots of places.